client/src/components/Challenge/AddChallenge/index.jsx

```diff
@@ -11,24 +11,57 @@ class AddChallenge extends Component {
   state = { 
     title: '',
     content: '',
-    difficulty: null
+    difficulty: null,
+    fn: '',
+    input: '',
+    output: '',
+    testcode: [],
    }
 
   submitChallenge = async (e) => {
     e.preventDefault();
-    const { title, content, difficulty } = this.state;
+    if (this.state.testcode.length === 0) {
+      return alert('Please add tests cases before submit');
+    }
+    const { title, content, difficulty, fn, testcode} = this.state;
     const id = localStorage.getItem('id');
+    
     const body = {
       title,
       content,
       difficulty,
+      fn,
       user_id: id,
       type: 0
     }
+    
     const result = await axios.post('http://localhost:3396/api/challenges/addChallenge', body);
+    this.state.testcode.forEach( async (test) => {
```

Comment:
pretty sure async doesn't work as expected in forEach functions, also this doesn't need to be asynchronous.